sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageId.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import java.util.Objects;
+
+/**
+ * This represents Message id in various forms.
```

Comment:
Add more details about what are the various forms of message id and why and how to use this.